src/main/java/seedu/address/model/recipe/attribute/Ingredient.java

```diff
@@ -3,67 +3,74 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.util.List;
+
 /**
- * Represents a Recipe's ingredient in its {@code IngredientList}.
- * Guarantees: immutable; is valid as declared in {@link #isValidIngredient(String)}
+ * Represents a Recipe's ingredient in its {@code IngredientList}. Guarantees:
+ * immutable; is valid as declared in {@link #isValidIngredient(String)}
  */
 public class Ingredient {
 
-    public static final String MESSAGE_CONSTRAINTS = "Ingredient can take any values, and it should not be blank";
-
-    /*
-     * The first character of the ingredient must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
-     */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String MESSAGE_CONSTRAINTS = "Each ingredient should be of the form:"
+            + "'ing/ing1, ing1qty; ing2, ing2qty;...'."
+            + "Spaces are optional. Valid ingredient quantities are specified in the User Guide. ";
 
     public final String name;
     private String quantity;
+    private Quantity quantity2;
 
     /**
-     * Constructs an {@code Ingredient}.
+     * Construct an Ingredient from {@code ingredientString}; the parameter is
+     * required to be non-null, and must follow a pattern.
      *
-     * @param name A valid ingredient string.
+     * @param ingredientString the {@link String} to be decoded
      */
-    public Ingredient(String name, String quantity) {
-        requireNonNull(name);
-        requireNonNull(quantity);
-        checkArgument(isValidIngredient(name), MESSAGE_CONSTRAINTS);
-        checkArgument(isValidIngredient(quantity), MESSAGE_CONSTRAINTS);
-        this.name = name;
-        this.quantity = quantity;
+    public Ingredient(String ingredientString) {
```

Comment:
As I've written the classes now, I've progressively 'subcontracted' the `Ingredient`, `Quantity` and `Unit` parsing to each of the respective classes.

In other words, the constructor for each of the above takes a string and all the parsing happens inside the constructor. Should I pull that behaviour out into the `util.parser` package, or leave it as is? 

I wrote it this way to abstract away the parsing, so that you can just pass in a string and everything happens inside the class, don't need to worry about parsing outside of it.

Reply:
I think this way should work, at least for now 👍

Reply:
Yes, for now its fine. But later on u will need to put it under parser package as all our parsing happens in parser package.